profiles/api/oauth2.py

```diff
@@ -118,21 +118,20 @@ def _token() -> Response:
 
         json_data = json.loads(response.text)
 
-        if 'access_token' not in json_data:
-            raise ValueError('No access_token')
-        elif 'expires_in' not in json_data:
-            raise ValueError('No expires_in')
-        elif json_data.get('token_type').lower() != 'bearer':
-            raise ValueError('Got token_type {}, expected Bearer'.format(
-                json_data.get('token_type')))
-
         json_data = {key: json_data[key] for key in
                      ['access_token', 'expires_in', 'name', 'orcid', 'token_type']}
 
+        if json_data.get('token_type').lower() != 'bearer':
+            raise ValueError('Got token_type {}, expected Bearer'.format(
+                json_data.get('token_type')))
+
         try:
             profile = profiles.get_by_orcid(json_data['orcid'])
-            profile.name = Name(json_data['name'])
+            if json_data.get('name', ''):
+                profile.name = Name(json_data['name'])
         except ProfileNotFound:
+            if not json_data.get('name', ''):
+                raise InvalidRequest('No name visible')
```

Comment:
I'm guessing this is the right type of error. Journal will need to be able to display a friendly error message.